frontend/app/assets/javascripts/spree/frontend/views/spree/products/index.js

```diff
@@ -108,4 +108,24 @@ Spree.ready(function ($) {
       setNewUrl(searchParams);
     }
   });
+
+  $('#filterPriceRange').on('submit', function(event) {
+    event.preventDefault();
+
+    var $priceRangeForm = $(event.currentTarget);
+    var minPrice = parseInt($priceRangeForm.find('[name="min_price"]').val());
+    var maxPrice = parseInt($priceRangeForm.find('[name="max_price"]').val());
+
+    var searchParams = new URLSearchParams(location.search);
+    var priceRange = [
+      SPREE_CURRENCY_SYMBOL,
+      minPrice,
+      ' - ',
+      SPREE_CURRENCY_SYMBOL,
+      maxPrice
+    ]
```

Comment:
@mad-eel we can utilize http://openexchangerates.github.io/accounting.js/ for this, we already have that library in frontend

Reply:
https://github.com/spree/spree/blob/c47e591882d79e766847c627773d29186aa2224b/frontend/app/assets/javascripts/spree/frontend/checkout/shipment.js#L36